src/Core/Configurations/JsonConfigSchemaVaidator.cs

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO.Abstractions;
+using System.Reflection;
+using Azure.DataApiBuilder.Config;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Microsoft.Extensions.Logging;
+using NJsonSchema;
+using NJsonSchema.Validation;
+
+namespace Azure.DataApiBuilder.Core.Configurations;
+
+public class JsonConfigSchemaValidator
+{
+    private static ILogger<JsonSchemaValidator>? _logger = new LoggerFactory().CreateLogger<JsonSchemaValidator>();
+    private static IFileSystem _fileSystem = new FileSystem();
+
+    /// <summary> 
+    /// Sets the logger and file system for the JSON config schema validator. 
+    /// </summary> 
+    /// <param name="jsonSchemaValidatorLogger">The logger to use for the JSON schema validator.</param> 
+    /// <param name="fileSystem">The file system to use for the JSON schema validator.</param>
+    public static void SetLoggerAndFileSystemForJsonConfigSchemaValidator(
+        ILogger<JsonSchemaValidator> jsonSchemaValidatorLogger,
+        IFileSystem fileSystem)
+    {
+        _logger = jsonSchemaValidatorLogger;
+        _fileSystem = fileSystem;
+    }
+
+     /// <summary> 
+    /// Validates a JSON schema against JSON data. 
+    /// </summary> 
+    /// <param name="jsonSchema">The JSON schema to validate against.</param> 
+    /// <param name="jsonData">The JSON data to validate.</param> 
+    /// <returns>A tuple containing a boolean indicating
+    /// if the validation was successful and a collection of validation errors if there were any.</returns> 
+    public static async Task<(bool ,ICollection<ValidationError>?)> ValidateJsonConfigWithSchemaAsync(string? jsonSchema, string? jsonData)
+    {
+        try{
+            JsonSchema schema = await JsonSchema.FromJsonAsync(jsonSchema);
+            ICollection<ValidationError> validationErrors = schema.Validate(jsonData, SchemaType.JsonSchema);
+
+            if (!validationErrors.Any())
+            {
+                _logger!.LogInformation("The config satisfies the schema requirements.");
+                return (true, null);
+            }
+            else
+            {
+                return (false, validationErrors);
+            }
+        }
+        catch (Exception e)
+        {
+            _logger!.LogError($"Failed to validate config against schema due to \n{e.Message}");
+            return (false, null);
+        }
+    }
+
+    /// <summary> 
+    /// Validates the given json config. 
+    /// </summary> 
+    /// <param name="jsonConfig">The JSON config file to validate.</param> 
+    public static async Task ValidateJsonConfig(string jsonConfig)
+    {
+        // deserialize jsonData to runtimeConfig
+        if (!RuntimeConfigLoader.TryParseConfig(jsonConfig, out RuntimeConfig? config))
```

Comment:
Cant we get the config from the RuntimeConfigProvider which could have it already parsedf? 
If CLI wont have it, can we do the Parsing only for the CLI portion when you implement `validate` but when its being called from Service, reuse the already parsed config object?